src/awkward/_broadcasting.py

```diff
@@ -57,20 +57,27 @@ class BroadcastOptions(TypedDict):
 
 
 def length_of_broadcast(inputs: Sequence) -> int | type[unknown_length]:
-    maxlen = 1
-
+    max_length: int | None = None
     has_seen_unknown_length: bool = False
     for x in inputs:
-        if isinstance(x, Content):
-            if x.length is unknown_length:
-                has_seen_unknown_length = True
-                continue
-            maxlen = max(maxlen, x.length)
+        if not isinstance(x, Content):
+            continue
+        if x.length is unknown_length:
+            has_seen_unknown_length = True
+        elif max_length is None:
+            max_length = x.length
+        else:
+            max_length = max(max_length, x.length)
 
     if has_seen_unknown_length:
-        return unknown_length
+        if max_length is None:
+            return unknown_length
+        else:
+            return max_length
+    elif max_length is None:
+        return 1
     else:
-        return maxlen
+        return max_length
```

Comment:
This code should now return a concrete length in the `unknown-length` - `int` case.